torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -516,9 +511,7 @@ def __init__(
             _check_orig_params_flattened(self, self._ignored_params)
             self._register_flat_param()
 
-        # TODO (revisit): I explicitly delete this because we do want to keep
-        # references to these from FSDP. I only added it to the state for
-        # convenience in this refactoring.
+        # Delete to avoid keeping references after the constructor
```

Comment:
We can revisit if we want to get around this `delattr` as a follow-up.

Reply:
This makes sense to me